lib/Pipes/PipelineManager.cpp

```diff
@@ -374,6 +374,39 @@ llvm::Error PipelineManager::produceAllPossibleTargets(bool ExpandTargets) {
   return llvm::Error::success();
 }
 
+const pipeline::Step::AnalysisValueType &
+PipelineManager::getAnalysis(const AnalysisReference &Reference) const {
+  auto &Step = getRunner().getStep(Reference.getStepName());
+
+  auto Analysis = llvm::find_if(Step.analyses(),
+                                [&](const Step::AnalysisValueType &Analysis)
+                                  -> bool {
+                                  return Analysis.first()
+                                         == Reference.getAnalysisName();
+                                });
+  return *Analysis;
+}
+
+llvm::Expected<DiffMap>
+PipelineManager::runAnalyses(const pipeline::AnalysesList &List,
+                             InvalidationMap &Map,
+                             const llvm::StringMap<std::string> &Options,
+                             llvm::raw_ostream *DiagnosticLog) {
+  auto Result = Runner->runAnalyses(List, Map, Options);
+  if (not Result)
+    return Result.takeError();
+
+  recalculateAllPossibleTargets();
+
+  // TODO: to remove once invalidations are working
+  if (auto Invalidations = invalidateAllPossibleTargets(); !!Invalidations)
```

Comment:
Mmmh, is this the same behavior of running an individual analysis?
Do we say "always everything invalidated" even if we run a single analysis?

Reply:
yeah, but this is to be dropped when the invalidation works correclty